tests/src/ComparisonUtils.h

```diff
@@ -140,11 +140,51 @@ VECTOR2_COMPARE(float, edm4hep::Vector2f)
 
 // Macro for comparing the return types of the different functions and return
 // false if they are not equal while also emitting a message
-#define ASSERT_COMPARE_VALS(lcioV, edm4hepV, msg)                                                 \
-  if ((lcioV) != (edm4hepV)) {                                                                    \
-    std::cerr << msg << " (LCIO: " << (lcioV) << ", EDM4hep: " << (edm4hepV) << ")" << std::endl; \
-    return false;                                                                                 \
+
+// Only enable for vectors
+template<typename LCIO, typename EDM4hep, typename = void>
+struct has_size_method : std::false_type {};
+
+template<typename LCIO, typename EDM4hep>
+struct has_size_method<LCIO, EDM4hep, std::void_t<decltype(std::declval<EDM4hep>().size())>> : std::true_type {};
+
+template<typename LCIO, typename EDM4hep>
+bool ASSERT_COMPARE_VALS_TEMPLATE(LCIO lcioV, EDM4hep edm4hepV, const std::string& msg)
```

Comment:
I just realized: This has to remain a macro that **only returns `false` in case of a comparison failure**. All the tests will abort early at the first successful test otherwise, as they assume that the macro will just do nothing in case of success.